### PR TITLE
Fix windows build

### LINKED ISSUE #31: Can't build the example with React
### Reproduction: 

1. Clone this repo: `git clone https://github.com/eirslett/storybook-builder-vite.git`
2. Go to the React example folder: `cd storybook-builder-vite/packages/example-react`
3. Install packages: `npm i` 
4. Try to build the storybook app: `npm run build-storybook` 💥🐛

### Logs: 
```node
🚀 npm run build-storybook

> example-react@0.0.7 build-storybook D:\Software\Github\storybook-builder-vite\packages\example-react
> build-storybook

info @storybook/react v6.3.0-rc.10
info
info => Cleaning outputDir: D:\Software\Github\storybook-builder-vite\packages\example-react\storybook-static
info => Loading presets
ERR! Addon value should end in /register OR it should be a valid preset https://storybook.js.org/docs/react/addons/writing-presets/
ERR! @storybook/addon-links
vite v2.3.7 building for production...
transforming (1154) node_modules\@popperjs\core\lib\dom-utils\isTableElement.jsConflicting namespaces: "node_modules\@storybook\client-api\dist\esm\index.js" re-exports "combineParameters" from both "node_modules\@storybook\client-api\dist\esm\index.js" and "node_modules\@storybook\client-api\dist\esm\parameters.js" (will be ignored)
Conflicting namespaces: "node_modules\@storybook\components\dist\esm\index.js" re-exports "components" from both "node_modules\@storybook\components\dist\esm\index.js" and "node_modules\@storybook\components\dist\esm\typography\DocumentFormatting.js" (will be ignored)
Conflicting namespaces: "node_modules\@storybook\addon-docs\dist\esm\blocks\index.js" re-exports "SourceState" from both "node_modules\@storybook\addon-docs\dist\esm\blocks\Canvas.js" and "node_modules\@storybook\addon-docs\dist\esm\blocks\Source.js" (will be ignored)
Use of eval is strongly discouraged, as it poses security risks and may cause issues with minification
Use of eval is strongly discouraged, as it poses security risks and may cause issues with minification
Use of eval is strongly discouraged, as it poses security risks and may cause issues with minification
✓ 1180 modules transformed.
rendering chunks (7)...[vite:build-html] The "fileName" or "name" properties of emitted files must be strings that are neither absolute nor relative paths, received "../../../../../../iframe.html".
ERR! Error: The "fileName" or "name" properties of emitted files must be strings that are neither absolute nor relative paths, received "../../../../../../iframe.html".
ERR!     at error (D:\Software\Github\storybook-builder-vite\packages\example-react\node_modules\rollup\dist\shared\rollup.js:164:30)
ERR!     at FileEmitter.emitFile (D:\Software\Github\storybook-builder-vite\packages\example-react\node_modules\rollup\dist\shared\rollup.js:12942:24)
ERR!     at Object.generateBundle (D:\Software\Github\storybook-builder-vite\packages\example-react\node_modules\vite\dist\node\chunks\dep-bc228bbb.js:24622:22)
ERR!     at async Bundle.generate (D:\Software\Github\storybook-builder-vite\packages\example-react\node_modules\rollup\dist\shared\rollup.js:13388:9)
ERR!     at async handleGenerateWrite (D:\Software\Github\storybook-builder-vite\packages\example-react\node_modules\rollup\dist\shared\rollup.js:20791:23)
ERR!     at async doBuild (D:\Software\Github\storybook-builder-vite\packages\example-react\node_modules\vite\dist\node\chunks\dep-bc228bbb.js:45406:20)
ERR!     at async build (D:\Software\Github\storybook-builder-vite\packages\example-react\node_modules\vite\dist\node\chunks\dep-bc228bbb.js:45230:16)
ERR!     at async build (D:\Software\Github\storybook-builder-vite\packages\example-react\node_modules\storybook-builder-vite\build.js:27:5)
ERR!     at async Promise.all (index 1)
ERR!     at async buildStaticStandalone (D:\Software\Github\storybook-builder-vite\packages\example-react\node_modules\@storybook\core-server\dist\cjs\build-static.js:141:28)
ERR!     at async buildStatic (D:\Software\Github\storybook-builder-vite\packages\example-react\node_modules\@storybook\core-server\dist\cjs\build-static.js:161:5)
ERR!  Error: The "fileName" or "name" properties of emitted files must be strings that are neither absolute nor relative paths, received "../../../../../../iframe.html".
ERR!     at error (D:\Software\Github\storybook-builder-vite\packages\example-react\node_modules\rollup\dist\shared\rollup.js:164:30)
ERR!     at FileEmitter.emitFile (D:\Software\Github\storybook-builder-vite\packages\example-react\node_modules\rollup\dist\shared\rollup.js:12942:24)
ERR!     at Object.generateBundle (D:\Software\Github\storybook-builder-vite\packages\example-react\node_modules\vite\dist\node\chunks\dep-bc228bbb.js:24622:22)
ERR!     at async Bundle.generate (D:\Software\Github\storybook-builder-vite\packages\example-react\node_modules\rollup\dist\shared\rollup.js:13388:9)
ERR!     at async handleGenerateWrite (D:\Software\Github\storybook-builder-vite\packages\example-react\node_modules\rollup\dist\shared\rollup.js:20791:23)
ERR!     at async doBuild (D:\Software\Github\storybook-builder-vite\packages\example-react\node_modules\vite\dist\node\chunks\dep-bc228bbb.js:45406:20)
ERR!     at async build (D:\Software\Github\storybook-builder-vite\packages\example-react\node_modules\vite\dist\node\chunks\dep-bc228bbb.js:45230:16)
ERR!     at async build (D:\Software\Github\storybook-builder-vite\packages\example-react\node_modules\storybook-builder-vite\build.js:27:5)
ERR!     at async Promise.all (index 1)
ERR!     at async buildStaticStandalone (D:\Software\Github\storybook-builder-vite\packages\example-react\node_modules\@storybook\core-server\dist\cjs\build-static.js:141:28)
ERR!     at async buildStatic (D:\Software\Github\storybook-builder-vite\packages\example-react\node_modules\@storybook\core-server\dist\cjs\build-static.js:161:5) {
ERR!   code: 'PLUGIN_ERROR',
ERR!   pluginCode: 'VALIDATION_ERROR',
ERR!   plugin: 'vite:build-html',
ERR!   hook: 'generateBundle'
ERR! }
```

### System Info:

```node
  System:
    OS: Windows 10 10.0.19041
    CPU: (8) x64 Intel(R) Core(TM) i7-7700HQ CPU @ 2.80GHz
    Memory: 4.82 GB / 15.87 GB
  Binaries:
    Node: 12.22.1 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.5 - C:\Program Files (x86)\Yarn\bin\yarn.CMD
    npm: 6.14.12 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Edge: Spartan (44.19041.964.0), Chromium (91.0.864.41)
    Internet Explorer: 11.0.19041.1
  npmPackages:
    vite: ^2.3.7 => 2.3.7
```

### PR DESCRIPTION
**Pull request by schalk-b**
Fixes #31

### CODE REVIEW

**Comment by eirslett** [comment]
Looks like there's a regression - now the Ubuntu build fails?

File: packages/storybook-builder-vite/code-generator-plugin.js [hunk 1]
```diff
@@ -1,10 +1,12 @@
-const fs = require('fs');
 const path = require('path');
 const { transformIframeHtml } = require('./transform-iframe-html');
 const { generateIframeScriptCode } = require('./codegen-iframe-script');
+const { normalizePath } = require('vite');
 
 module.exports.codeGeneratorPlugin = function codeGeneratorPlugin(options) {
     const virtualFileId = '/virtual:/@storybook/builder-vite/vite-app.js';
+    const iframeAbsoluteFileName = path.resolve(__dirname, 'input', 'iframe.html')
+    const iframeRelativeFileName = normalizePath(path.relative(process.cwd(), iframeAbsoluteFileName))
```
Comment (eirslett) [inline packages/storybook-builder-vite/code-generator-plugin.js:9]: Can we prevent `process.cwd()` usage here? @shilman I think it's a design choice in Storybook that directories are configurable, and not based on working directories?

**Comment by schalk-b** [comment]
> Looks like there's a regression - now the Ubuntu build fails?

Sorry, I didn't have a change to resolve the regression issue. I've looked into it further and I'm hitting a roadblock.

https://github.com/vitejs/vite/blob/main/packages/vite/src/node/plugins/html.ts#L385
`config.root` = `C:/Users/schal/storybook-builder-vite/packages/example-react`
`id` = `C:/Users/schal/storybook-builder-vite/packages/storybook-builder-vite/input/iframe.html`
`shortEmitName` = `../storybook-builder-vite/input/iframe.html`

However, vite:html complains at this point because it expects an absolute path relative to `config.root`. I have a feeling your originally approach was correct and using a *virtual* input is the way to go.

File: packages/storybook-builder-vite/code-generator-plugin.js (hunk 1 above)
Comment (shilman) [reply packages/storybook-builder-vite/code-generator-plugin.js:9]: Yes there's a `--configDir` flag or something like that

**Comment by schalk-b** [comment]
I've re-written the fix to still use the `virtual` input entry, however it is now using `${config.root}/iframe.html` for the virtual id.

**Comment by schalk-b** [comment]
Did a squash force push to tidy things up

**Comment by IanVS** [comment]
I rebased on top of https://github.com/eirslett/storybook-builder-vite/pull/51, and CI passes now.  This looks like a good fix, thanks @schalk-b!

**Pull request merged**